presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,548 @@ Tuning Presto
+
+
+``node-scheduler.min-candidates``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``10``
+ * **Description:** The minimal number of nodes candidates proposed by scheduler to do every job in system. Setting this allows to manipulate global parallelism. The higher value is recommended for system having lots of nodes and small number of huge queries. The lower value is recommended in system that have higher number of smaller queries. Also this setting is connected with ``node-scheduler.network-topology`` - while using ``flat`` it is important to align this value with number of nodes that backend required for queries is split between (or higher).
+
+
+``node-scheduler.multiple-tasks-per-node-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** Allow nodes to be selected multiple times by the node scheduler, in a single stage. With that property set to ``false`` the ``node-scheduler.min-candidates`` is capped at number of nodes in system. Having this set may allow better scheduling and concurrency reducing number of outliers and possibly speeding up computations. Also it may allow to collect smaller bulks of result in unstable network conditions. The drawbacks are that some optimization may work less efficiently on smaller partitions. Also slight hardware efficiency drop is expected in heavy loaded system.
+
+.. _node-scheduler-network-topology:
+
+``node-scheduler.network-topology``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (``legacy`` or ``flat``)
+ * **Default value:** ``legacy``
+ * **Description:** Sets the network topology to use when scheduling splits. ``legacy`` will ignore the topology when scheduling splits. ``flat`` will try to schedule splits on the same host as the data is located by reserving 50% of the work queue for local splits. It is recommended to use ``flat`` for clusters where distributed storage runs on same nodes as presto workers.
+
+
+.. _tuning-pref-optimizer:
+
+Optimizer properties
+--------------------
+
+``optimizer.processing-optimization``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (``disabled``, ``columnar`` or ``columnar_dictionary``)
+ * **Default value:** ``disabled``
+ * **Description:** Serves as default value for ``processing_optimization`` session property. Setting this property will allow to set up optimization for filtering operators. Setting ``columnar`` allows to use columnar processing instead of using row by row to process the filter. Setting ``columnar_dictionary`` adds additional dictionary to simplify columnar scan. Setting this to value other than ``disabled`` is advised for data containing large rows often filtered by simple key.
```

Comment:
Setting this property changes how filtering operators are processed.  Setting it to ``columnar`` allows Presto to use columnar processing instead of row by row. Setting ``columnar_dictionary`` adds additional dictionary to simplify columnar scan. Setting this to a value other than ``disabled`` may improve performance for data containing large rows often filtered by a simple key.  This can also be specified on a per-query basis using the ``processing_optimization`` session property.